calyx-frontend/src/attributes.rs

```diff
@@ -145,3 +147,13 @@ impl Attributes {
             .join(sep)
     }
 }
+
+#[cfg(feature = "serialize")]
+impl serde::Serialize for HeapAttrInfo {
```

Comment:
Seems like a reasonable place to put it! We don't really want to expose the internal details of `HeapAttrInfo` to users.